test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -41,36 +41,77 @@ class XdsOverrideHostTest : public LoadBalancingPolicyTest {
          Json::Object{{"childPolicy", Json::Array{{child_policy_config}}}}}}});
   }
 
+  absl::optional<RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>>
+  InitRoundRobin(absl::Span<const absl::string_view> addresses) {
+    absl::optional<RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>>
+        maybe_picker;
+    EXPECT_EQ(ApplyUpdate(BuildUpdate(addresses, MakeXdsOverrideHostConfig()),
+                          policy_.get()),
+              absl::OkStatus());
+    ExpectConnectingUpdate();
+    RefCountedPtr<LoadBalancingPolicy::SubchannelPicker> picker;
+    for (size_t i = 0; i < addresses.size(); ++i) {
+      auto* subchannel = FindSubchannel(addresses[i]);
+      EXPECT_NE(subchannel, nullptr);
+      if (subchannel == nullptr) {
+        return absl::nullopt;
+      }
+      EXPECT_TRUE(subchannel->ConnectionRequested());
+      subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
+      subchannel->SetConnectivityState(GRPC_CHANNEL_READY);
+      if (i == 0) {
+        auto picker = WaitForConnected();
+        ExpectRoundRobinPicks(picker.get(), {addresses[0]});
+      } else {
+        maybe_picker = WaitForRoundRobinListChange(
+            absl::MakeSpan(addresses).subspan(0, i),
+            absl::MakeSpan(addresses).subspan(0, i + 1));
+      }
+    }
+    return maybe_picker;
+  }
+
   OrphanablePtr<LoadBalancingPolicy> policy_;
 };
 
 TEST_F(XdsOverrideHostTest, DelegatesToChild) {
-  ASSERT_NE(policy_, nullptr);
-  const std::array<absl::string_view, 2> kAddresses = {"ipv4:127.0.0.1:441",
-                                                       "ipv4:127.0.0.1:442"};
-  EXPECT_EQ(policy_->name(), "xds_override_host_experimental");
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
   EXPECT_EQ(ApplyUpdate(BuildUpdate(kAddresses, MakeXdsOverrideHostConfig()),
```

Comment:
Might as well replace most of the body of this test with calling `InitRoundRobin()`, since they seem to be identical.

Reply:
Not sure what you mean. Have not found a function named `InitRoundRobin()`

Reply:
I was referring to the function that we renamed to `ExpectStartupWithRoundRobin()`.  What I'm saying is that the body of the `DelegatesToChild` test is basically duplicating the functionality of `ExpectStartupWithRoundRobin()`, so we can basically replace most of the body of the test with a call to that function.